source/web/components/JobEditor.js

```diff
@@ -268,23 +268,34 @@ export default class JobEditor extends Component {
                     />
                 </FormGroup>
                 <FormGroup label="Parents" labelFor="parents">
-                    <ControlGroup>
-                        <InputGroup
+                    <div className={`${Classes.SELECT} select-editable ${Classes.FILL}`}>
+                        <select
+                            onChange={evt => this.setState({ newParent: evt.target.value })}
+                            value={this.state.newParent}
+                        >
+                            <For each="jobId" index="idx" of={this.props.jobIds}>
+                                <option key={idx} value={jobId}>
```

Comment:
Key could be `jobId`, so that the order isn't important. `idx` can change too easily :)

Reply:
Fixed!